build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/rbs/client_class.rb

```diff
@@ -0,0 +1,176 @@
+        end
+
+        def operations
+          shapes = @api["shapes"]
+          @api["operations"].map do |name, body|
+            method_name = Underscore.underscore(name)
+            indent = " " * (12 + method_name.length)
+            input_shape_name = body.dig("input", "shape")
+            arguments = nil
+            include_required = false
+            if input_shape_name
+              input_shape = shapes[input_shape_name]
+              builder = AwsSdkCodeGenerator::RBS::KeywordArgumentBuilder.new(
+                api: @api,
+                shape: input_shape,
+                newline: true,
+              )
+              arguments = builder.format(indent: indent)
+              include_required = input_shape["required"]&.empty?&.!
+            end
+            if AwsSdkCodeGenerator::Helper.operation_streaming?(body, @api)
+              block = " ?{ (*untyped) -> void }"
+            end
+            if output_shape_name = body.dig("output", "shape")
+              output_shape = shapes[output_shape_name]
+              data = AwsSdkCodeGenerator::RBS.to_type(body.fetch( "output"), @api)
+              interface = "_#{name}ResponseSuccess"
+            else
+              output_shape = nil
+              data = "::Aws::EmptyStructure"
+              interface = empty_interface = "::Seahorse::Client::_ResponseSuccess[::Aws::EmptyStructure]"
+            end
+            returns_members = output_shape&.[]("members")&.inject([]) do |a, (member_name, member_ref)|
+              next a if SKIP_MEMBERS.include?(member_name)
+              a << {
+                method_name: Underscore.underscore(member_name),
+                returns: AwsSdkCodeGenerator::RBS.to_type(member_ref, @api),
+              }
+            end
+
+            {
+              method_name: method_name,
+              signature: AwsSdkCodeGenerator::RBS::MethodSignature.new(
+                method_name: method_name,
+                overloads: [
+                  "(#{arguments})#{block} -> #{interface}",
+                  "(#{include_required ? "" : "?"}Hash[Symbol, untyped] params, ?Hash[Symbol, untyped] options)#{block} -> #{interface}",
+                ]).signature,
+              interface: interface,
+              data: data,
+              returns_members: returns_members,
+              empty_structure: output_shape.nil?
+            }
+          end
+        end
+
+        def waiters?
+          @waiters.size > 0
+        end
+
+        def waiters_first
+          [@waiters.first]
+        end
+
+        def waiters_others
+          @waiters[1..-1]
+        end
+
+        private
+
+        def documented_plugin_options(plugins)
+          i = 0
+          plugins.map(&:options).flatten.select(&:documented?).sort_by do |opt|
+            [opt.required ? 'a' : 'b', opt.name, i += 1]
+          end
+        end
+
+        def build_keyword_arguments(plugins)
+          buffer = plugins.map do |opt|
+            type = case opt.name
```

Comment:
Its great to have these - but I think rather than hard coding in an option -> type here, we can add `rbs_type` to the [PluginOption](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/seahorse/client/plugin.rb#L110) and then in each plugin where these are defined we can specify the rbs.

I know this doesn't work for some of the net http and other  (eg http_proxy, ect) since they aren't specified by plugins, so we can leave those here.  But adding `rbs_type` to the plugins should make it easier for us to keep types up to date when we add new configuration....

Reply:
Great idea! I will give it a try.